### DISCUSSION TITLE
真的十分感谢这个项目的所有贡献者！

### DISCUSSION THREAD

**Discussion by EricYoung37** [XTLS/Xray-core#2261] [Show and tell]
我按照[小小白白话文](https://xtls.github.io/document/)完成了部署，测试之后成功了！

我完成自己的部署花费了2个晚上。其实我之前也尝试过V2Ray，但是花了3天都没有办法成功。
**个人**认为，V2Ray那边最大的问题是：
- 指南内容和软件版本的差距太大了。依照指南写配置文件不是很有用。我看了V2Ray和V2fly两个GitHub账户，但是Core的新旧指南以及Fly的网站的配置文件还是不太好用。
- GitHub Issues对新人比较严格，所以也不是很敢问问题（当然我觉得这个是因为我自己实力不够），但有意思的是我在配置XRay时并没有感觉需要去Issues发问题的时候，所以我觉得大概率还是指南的关系。而也正是在Issues里寻找问题解决方案时我看到有人建议Xray。才了解到Xray算是V2Ray的分支，因此决定尝试Xray。
- 配置文件的Examples已经至少2年没更新了，无法确定是否还有效。

当然并不是说V2Ray不好，V2Ray和Xray应该是有高度重合的贡献者和使用群体的，V2Ray的贡献者们也十分伟大。没办法成功使用V2Ray应该也是因为我自己实力仍有巨大不足。

决定尝试Xray后，我翻看了一下**Xray-core，Xray-examples，Xray的指南，更新的时间都很近**！即使是指南第7、8章也是今年初有更新的。这让我觉得成功率提高了。而事实正是如此。**真的十分感谢Xray各个部分的贡献者们！！！**

在Xray部署中，我大部分时间花在：
- 复习（学习）一些自己薄弱的Linux知识（最主要是用户权限和密码的管理那一部分）
- 对比其他相关文献来制定适合自己环境的方案。比如：我的VPS要额外设置开放端口的规则，了解和对比DNS供应商，对比acme.sh的GitHub仓库来写安装证书的指令，查找适用于Ubuntu的BBR开启方法等等。

指南作者的设计真的十分用心合理。即使最后涉及Xray本体的只有最后2章，前面那些准备工序也是十分有益的。这一部分对于有基础的人来说都是一个很不错的学习材料（涵盖了计算机网络、网页开发等领域）。作者完全可以不写证书或者网站搭建那一部分的，但是那样的话TLS和回落这一部分很重要的东西可能劝退新人。

之后有时间我应该会去研读一下[配置指南](https://xtls.github.io/config/#%E6%A6%82%E8%BF%B0)，再之后尝试一下不同的配置文件组合。

再次感谢Xray的所有贡献者！

**Comment by chika0801** [comment]
你想学xray配置还推荐以下：

https://www.v2fly.org/config/dns.html#dns-%E5%A4%84%E7%90%86%E6%B5%81%E7%A8%8B
https://guide.v2fly.org/basics/dns.html
这2个链接主要讲DNS模块工作流程，他们中提到的参数基本上都在Xray中使用。Xray文档没有第1个链接详细，我也是主要学的第1个链接的内容。

~https://toutyrater.github.io/~ 已过时，供参考。

综合的dns流程介绍。这个教程日期虽然很旧，但是文章写的很好，可能参数和现在Xray的有点点对不上，大部分参数是对得上的。

它先以http socks代理环境举例，再讲到比如客户端本机把dns服务器设置成127.0.0.1，再用自由门入站导入到Xray处理，最后还提到透明代理环境用Fakedns的一点点内容。也是入门精选。

https://tachyondevel.medium.com/%E6%BC%AB%E8%B0%88%E5%90%84%E7%A7%8D%E9%BB%91%E7%A7%91%E6%8A%80%E5%BC%8F-dns-%E6%8A%80%E6%9C%AF%E5%9C%A8%E4%BB%A3%E7%90%86%E7%8E%AF%E5%A2%83%E4%B8%AD%E7%9A%84%E5%BA%94%E7%94%A8-62c50e58cbd0

~上面链接写的是以v2ray配置，实际就是现在Xray用的。~

~另外你有TG加Xray的群，那里客服也挺多的。~

**Reply by RPRX** [reply]
我看了下 https://toutyrater.github.io/ 的内容有些过时了（Victoria Raymond 时期的。。。），建议看比较新的文档、白话文

其它的都是 DNS？~~快进到 DNS 泄漏~~
